### PR TITLE
[ci-visibility] Only run git commands when `mocha` plugin is used and add benchmark test

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Related: https://github.com/DataDog/dd-trace-js/pull/2198 

* Only run git commands when `mocha` is being used (where they are used)
* Run a benchmark test and upload a datadog metric to keep track of the overhead 

### Motivation
* Do not run unnecessary git commands 
* Control the overhead added by our instrumentation when using `mocha`

### CODE REVIEW

File: packages/datadog-plugin-mocha/src/index.js [hunk 1]
```diff
@@ -156,6 +154,19 @@ class MochaPlugin extends Plugin {
     })
   }
 
+  addMochaSub (channelName, handler) {
+    this.addSub(channelName, (...args) => {
+      if (!this._hasBeenUsed) {
```
Comment (rochdev) [inline packages/datadog-plugin-mocha/src/index.js:159]: This could be done in `configure` instead.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-plugin-mocha/src/index.js (hunk 1 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-mocha/src/index.js:159]: I don't understand. You mean to include the `_hasBeenUsed` logic in the plugin's `configure` ? `configure` is called  even if the plugin is not used, so it wouldn't work, unless I'm missing something

Comment (rochdev) [reply packages/datadog-plugin-mocha/src/index.js:159]: I'm close to finishing a change that will lazy load plugins only when they are needed, which will make this easier to implement. It should be ready next week.

**Comment by juan-fernandez** [comment]
closing this in favor of https://github.com/DataDog/dd-trace-js/pull/2234. As for the benchmark test we can do it at a different PR

**Pull request closed without merging**